modules/util/util.js

```diff
@@ -97,15 +97,22 @@ export function utilGetAllNodes(ids, graph) {
 export function utilDisplayName(entity) {
     var localizedNameKey = 'name:' + utilDetect().locale.toLowerCase().split('-')[0];
     var name = entity.tags[localizedNameKey] || entity.tags.name || '';
-    var network = entity.tags.cycle_network || entity.tags.network;
 
     if (!name && entity.tags.ref) {
         name = entity.tags.ref;
+        var network = entity.tags.cycle_network || entity.tags.network;
         if (network) {
             name = network + ' ' + name;
         }
     }
 
+    if (!name && entity.tags['addr:housenumber']) {
+        name = entity.tags['addr:housenumber'];
+        if (entity.tags['addr:street']) {
+            name += ' ' + entity.tags['addr:street'];
```

Comment:
In some countries, the street name comes before the house number. Would it be feasible to use the address-formats.json as a guide to how to format these addresses?